Specify function return types:

error: type specifier missing, defaults to 'int'; ISO C99 and later do not support implicit int [-Wimplicit-int]

Include the correct headers for exit and memcmp:

error: implicitly declaring library function 'exit' with type 'void (int) __attribute__((noreturn))' [-Werror,-Wimplicit-function-declaration]
error: implicitly declaring library function 'memcmp' with type 'int (const void *, const void *, unsigned long)' [-Werror,-Wimplicit-function-declaration]

Include the correct headers for checking for intXX_t, u_intXX_t, and uintXX_t types:

error: expected ';' after expression
error: use of undeclared identifier 'uint16_t'
error: use of undeclared identifier 'c'
error: expected ';' after expression
error: use of undeclared identifier 'uint32_t'
error: use of undeclared identifier 'd'
error: use of undeclared identifier 'c'
error: use of undeclared identifier 'd'
--- configure.o	2001-05-10 11:10:49.000000000 -0500
+++ configure	2024-02-04 22:50:13.000000000 -0600
@@ -656,7 +656,7 @@
 #line 657 "configure"
 #include "confdefs.h"
 
-main(){return(0);}
+int main(){return(0);}
 EOF
 if { (eval echo configure:662: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   ac_cv_prog_cc_works=yes
@@ -1234,6 +1234,7 @@
 #line 1235 "configure"
 #include "confdefs.h"
 #include <ctype.h>
+#include <stdlib.h>
 #define ISLOWER(c) ('a' <= (c) && (c) <= 'z')
 #define TOUPPER(c) (ISLOWER(c) ? 'A' + ((c) - 'a') : (c))
 #define XOR(e, f) (((e) && !(f)) || (!(e) && (f)))
@@ -1479,7 +1480,7 @@
   cat > conftest.$ac_ext <<EOF
 #line 1481 "configure"
 #include "confdefs.h"
-main () {
+int main() {
   /* Are we little or big endian?  From Harbison&Steele.  */
   union
   {
@@ -1660,7 +1661,8 @@
 #line 1661 "configure"
 #include "confdefs.h"
 #include <stdio.h>
-main()
+#include <stdlib.h>
+int main()
 {
   FILE *f=fopen("conftestval", "w");
   if (!f) exit(1);
@@ -1699,7 +1701,8 @@
 #line 1700 "configure"
 #include "confdefs.h"
 #include <stdio.h>
-main()
+#include <stdlib.h>
+int main()
 {
   FILE *f=fopen("conftestval", "w");
   if (!f) exit(1);
@@ -1738,7 +1741,8 @@
 #line 1739 "configure"
 #include "confdefs.h"
 #include <stdio.h>
-main()
+#include <stdlib.h>
+int main()
 {
   FILE *f=fopen("conftestval", "w");
   if (!f) exit(1);
@@ -1777,7 +1781,8 @@
 #line 1778 "configure"
 #include "confdefs.h"
 #include <stdio.h>
-main()
+#include <stdlib.h>
+int main()
 {
   FILE *f=fopen("conftestval", "w");
   if (!f) exit(1);
@@ -1816,7 +1821,8 @@
 #line 1817 "configure"
 #include "confdefs.h"
 #include <stdio.h>
-main()
+#include <stdlib.h>
+int main()
 {
   FILE *f=fopen("conftestval", "w");
   if (!f) exit(1);
@@ -1907,7 +1913,7 @@
 cat > conftest.$ac_ext <<EOF
 #line 1909 "configure"
 #include "confdefs.h"
-#include <sys/types.h>
+#include <stdint.h>
 int main() {
 int16_t a; int32_t b; a = 1235; b = 1235;
 ; return 0; }
@@ -1936,7 +1942,7 @@
 cat > conftest.$ac_ext <<EOF
 #line 1938 "configure"
 #include "confdefs.h"
-#include <sys/types.h>
+#include <stdlib.h>
 int main() {
 u_int16_t c; u_int32_t d; c = 1235; d = 1235;
 ; return 0; }
@@ -2006,7 +2012,7 @@
 cat > conftest.$ac_ext <<EOF
 #line 2008 "configure"
 #include "confdefs.h"
-#include <sys/types.h>
+#include <stdint.h>
 int main() {
 uint16_t c; uint32_t d; c = 1235; d = 1235;
 ; return 0; }
@@ -2065,7 +2071,9 @@
 #line 2066 "configure"
 #include "confdefs.h"
 
-main()
+#include <stdlib.h>
+#include <string.h>
+int main()
 {
   char c0 = 0x40, c1 = 0x80, c2 = 0x81;
   exit(memcmp(&c0, &c2, 1) < 0 && memcmp(&c1, &c2, 1) < 0 ? 0 : 1);
